lib/ordered_map.h

```diff
@@ -124,6 +124,20 @@ class ordered_map {
     size_type max_size() const noexcept { return data_map.max_size(); }
     bool operator==(const ordered_map &a) const { return data == a.data; }
     bool operator!=(const ordered_map &a) const { return data != a.data; }
+    bool operator<(const ordered_map &a) const {
```

Comment:
This implementation was copied from `ordered_set.h`. Maybe there is a better way to do this.

Reply:
I would also add a comment that `data_map` here is ordered by `operator<` of underlying key type (via some crazy indirection, but still), so we are good on iteration here.